### PR TITLE
feat: Auto reconcile after new draft voucher submission in reco tool

### PR DESCRIPTION
**Pull request by marination**
- On "Edit in Full Page", create a new voucher in a new tab and listen to new voucher doc updates
- If voucher is deleted after creation, turn off listener
- If simply updated (saved), do nothing
- If submitted, reconcile with current transaction and update UI
- **If user moves to next transaction**, turn off listener. We don't want random transactions getting reconciled

> These actions will take place on a transaction given that it is the active selection.

Magic 🪄

### CODE REVIEW

**Comment by marination** [comment]
To fix:

- [ ] Check perms in APIs (separate PR)
- [x] Add type hints
- [x] Make sure API return values are predictable

File: banking/klarna_kosma_integration/doctype/bank_reconciliation_tool_beta/bank_reconciliation_tool_beta.py [hunk 1]
```diff
@@ -215,16 +207,33 @@ def create_payment_entry_bts(
...
 
 	payment_entry.submit()
+
+	return reconcile_entry(
+		bank_transaction_name, paid_amount, "Payment Entry", payment_entry.name
+	)
+
+
+@frappe.whitelist()
+def reconcile_entry(transaction_name, amount, voucher_type, voucher_name):
```
Comment (barredterra) [inline banking/klarna_kosma_integration/doctype/bank_reconciliation_tool_beta/bank_reconciliation_tool_beta.py]:
```suggestion
def reconcile_voucher(transaction_name, amount, voucher_type, voucher_name):
```

Let's make the function name match the parameter names?

File: banking/public/js/bank_reconciliation_beta/actions_panel/create_tab.js [hunk 2]
```diff
@@ -32,10 +32,23 @@ erpnext.accounts.bank_reconciliation.CreateTab = class CreateTab {
 	}
 
 	edit_in_full_page() {
+		var me = this;
```
Comment (barredterra) [inline banking/public/js/bank_reconciliation_beta/actions_panel/create_tab.js]: IIRC, `me === this` inside an arrow function.

File: banking/public/js/bank_reconciliation_beta/actions_panel/create_tab.js [hunk 3]
```diff
@@ -89,6 +102,41 @@ erpnext.accounts.bank_reconciliation.CreateTab = class CreateTab {
...
+						frappe.realtime.off("doc_update");
+						return;
+					}
+
+					me.actions_panel.after_transaction_reconcile(
+						response.message, true, doctype
+					);
+				}
+			}
+		})
```
Comment (barredterra) [inline banking/public/js/bank_reconciliation_beta/actions_panel/create_tab.js]:
```suggestion
		});
```

**Comment by barredterra** [comment]
Functionality LGTM

**Pull request merged**